Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -607,3 +627,17 @@ fileprivate extension TokenSyntax {
     }
   }
 }
+
+fileprivate extension SyntaxProtocol {
+  /// Returns this node or the first ancestor that satisfies `condition`.
```

Comment:
We have in SwiftParserDiagnostics and here now. Worth just adding in general?

Reply:
I don’t feel happy enough with this API to make it public yet. So I prefer to just have two copies of it for now.